### PR TITLE
Taking a crack at code digest idea.

### PR DESCRIPTION
**Pull request by gigamonkey**
Here's an implementation of the idea I was thinking of. For me it's easier to copy the original code and run `pbpaste | java CodeDigest` and then stick the resulting hash into the unit test than to convert a bunch of code into a string. And there's the possibility I mentioned of being able to automatically detect places where the code and the hash has diverged.

What do you think?

(Sorry about some of the gratuitous white space changes—my editor is set up to trim trailing whitespace. Which in the long run minimizes diffs but in cases like this causes larger diffs.)

### CODE REVIEW

**Review by bhoffman0 — APPROVED** [review]
I guess it doesn't hurt to include this, as another option.
I think I'll make it live after the PD week just in case though.

File: CodeTestHelper.java [hunk 1]
```diff
@@ -1452,6 +1456,53 @@ public boolean codeChanged(String origCode, boolean expected) {
...
+        try {
+            String msg = "Checking that code has been changed";
+            String digest = codeDigest(getCode());
+            boolean passed = !digest.equals(originalDigest);
+            results += formatOutput("true", "" + passed, msg, passed);
+            results += "\n";
+            return passed;
+        } catch (NoSuchAlgorithmException nsae) {
+            String msg = "Test could not be completed";
+            results += formatOutput("true", "false", msg, false);;
```
Comment (bhoffman0) [inline CodeTestHelper.java:1482]:
```suggestion
            results += formatOutput("true", "false", msg, false);
```

File: CodeTestHelper.java [hunk 2]
```diff
@@ -1333,7 +1337,7 @@ public String getCodeWithoutComments() {
         return code;
     }
 
-    public String removeComments(String code) {
+    public static String removeComments(String code) {
```
Comment (bhoffman0) [inline CodeTestHelper.java:1340]: This wouldn't affect anything in the current tests, would it?

Comment (gigamonkey) [reply CodeTestHelper.java:1340]: It shouldn't. It's only actually used within this class (and probably should be private).

**Pull request merged**